uwblueprint/extend-a-family
# Add route, service, and functionality for Admin Invitations

## Notion ticket link

[Invite Admin](https://www.notion.so/uwblueprintexecs/Invite-Admin-8b898b398018424dab08f27cc6f60adc)



## Implementation description
* Route POST endpoint added to authRoutes.ts
* Checks for request authorization and request integrity through middleware
* Creates new user with temporary password generated by 'generate-password' npm library
* Sends email to prompt for user account verification


## Steps to test
1. Use Postman to attempt to post to `/auth/inviteAdmin`. Get the bearer token from the frontend (details in Notion ticket).
2. Should receive email and be able to log in as new username and the temporary password given in the email.



## What should reviewers focus on?
* Making sure unauthorized and bad requests are caught as errors by the backend


## Checklist
